### PR TITLE
Component Deployment PuPr Hotfix

### PR DESCRIPTION
**Pull request by TajiHarrisMicrosoft**
# Description

Makes the following hotfix changes to unblock release:
- Puts experimental tag on ModelBatchDeployment and PipelineComponentBatchDeployment
- adds PipelineComponentBatchDeployment as _begin_create_or_update return option
- adds ModelBatchDeployment and deployment type option for _begin_create_or_update

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/model_batch_deployment.py [hunk 1]
```diff
@@ -18,11 +18,13 @@
 from azure.ai.ml.exceptions import ErrorCategory, ErrorTarget, ValidationErrorType, ValidationException
 from azure.ai.ml.constants._common import BASE_PATH_CONTEXT_KEY, PARAMS_OVERRIDE_KEY
 from azure.ai.ml.entities._util import load_from_dict
+from azure.ai.ml._utils._experimental import experimental
 from azure.ai.ml._schema._deployment.batch.model_batch_deployment import ModelBatchDeploymentSchema
 from .model_batch_deployment_settings import ModelBatchDeploymentSettings
 from .code_configuration import CodeConfiguration
 
 
+@experimental
```
Comment (needuv) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/model_batch_deployment.py:27]: does `ModelBatchDeploymentSettings` also need to experimental or is it already marked as such?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py [hunk 2]
```diff
@@ -39,7 +39,7 @@
 
 ops_logger = OpsLogger(__name__)
 logger, module_logger = ops_logger.package_logger, ops_logger.module_logger
-
+DeploymentType = TypeVar("DeploymentType", bound=Union[BatchDeployment, PipelineComponentBatchDeployment])
```
Comment (nancy-mejia) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py:42]: Do we need to update this to include `ModelBatchDeployment`?

Comment (TajiHarrisMicrosoft) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_batch_deployment_operations.py:42]: No, given that ModelBatchDeployment and BatchDeployment are follow the same creation flow

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/model_batch_deployment.py (hunk 1 above)
Comment (TajiHarrisMicrosoft) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/model_batch_deployment.py:27]: Good catch, will make that change as well

**Review by nancy-mejia — APPROVED** [review]

**Pull request merged**